File: src/main/java/com/pauloelienay/slinky/generics/IGenericEntity.java
package com.pauloelienay.slinky.generics;

import java.io.Serializable;

import static java.util.Objects.isNull;

@SuppressWarnings({"unused"})
public interface IGenericEntity<T, S extends Serializable> {
	S getId();

	default boolean isUpdate() {
		return isNull(getId());
	}
}

File: src/main/java/com/pauloelienay/slinky/generics/IGenericEntity.kt
package com.pauloelienay.slinky.generics

import org.springframework.beans.BeanUtils

interface IGenericEntity<S> {
    var id: S?

    val isUpdate: Boolean
        get() = id == null

    fun <T> copyAllButId(target: T): T {
        BeanUtils.copyProperties(this, target as Any, "id")
        return target
    }
}